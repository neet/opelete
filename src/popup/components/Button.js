import React from 'react';
import PropTypes from 'prop-types';

export default class Button extends React.PureComponent {

  static propTypes = {
    className: PropTypes.string,
    text: PropTypes.string.isRequired,
    href: PropTypes.string,
    children: PropTypes.node,
    onClick: PropTypes.func,
  }

  handleClick = () => {
    if ( this.props.href ) {
      window.location.href = this.props.href;
    } else if ( this.props.onClick ) {
      this.props.onClick();
    }
  }

  render() {
    const { text, children, className } = this.props;

    return (
      <button className={`button ${className}`} aria-label={text} onClick={this.handleClick}>
        { children || text }
      </button>
    );
  }

}
