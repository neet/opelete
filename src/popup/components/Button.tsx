import React from 'react';

export interface Props {
  className?: string;
  text?: string;
  href?: string;
  children?: React.ReactNode;
  onClick?: () => void;
}

export class Button extends React.PureComponent<Props> {

  private handleClick = () => {
    if ( this.props.href ) {
      window.location.href = this.props.href;
    } else if ( this.props.onClick ) {
      this.props.onClick();
    }
  }

  public render () {
    const { text, children, className } = this.props;

    return (
      <button className={`button ${className}`} aria-label={text} onClick={this.handleClick}>
        {children || text}
      </button>
    );
  }

}
