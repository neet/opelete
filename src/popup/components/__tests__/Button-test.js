import React from 'react';
import Button from '../Button';
import renderer from 'react-test-renderer';

describe('Button', () => {
  it('renders a button with link', () => {
    const tree = renderer.create(
      <Button
        className='foo'
        text='hoge'
        href='https://example.com'
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders a button with function', () => {
    const handler = jest.fn();
    const tree = renderer.create(
      <Button
        className='foo'
        text='hoge'
        onClick={handler}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
