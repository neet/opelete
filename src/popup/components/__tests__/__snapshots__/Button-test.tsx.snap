// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a button with function 1`] = `
<button
  aria-label="hoge"
  className="button foo"
  onClick={[Function]}
  title="hoge"
>
  hoge
</button>
`;

exports[`Button renders a button with link 1`] = `
<button
  aria-label="hoge"
  className="button foo"
  onClick={[Function]}
  title="hoge"
>
  hoge
</button>
`;
